src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+            return;
+        }
+        last.next = new Node<>(last, value, null);
+        last = last.next;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> newElement;
+        if (index == 0) {
+            newElement = new Node<>(null, value, first);
+            first = newElement;
+            newElement.next.prev = newElement;
+            size++;
+            return;
+        }
+        Node<T> element = first;
+        for (int i = 0; i < index; i++) {
+            element = element.next;
+        }
+        newElement = new Node<>(element.prev, value, element);
+        element.prev.next = newElement;
+        element.prev = newElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexNotExist(index);
+        Node<T> element = first;
+        for (int i = 0; i < index; i++) {
+            element = element.next;
+        }
+        return element == null ? null : element.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexNotExist(index);
+        Node<T> element = first;
+        Node<T> newElement;
+        if (index == 0) {
+            newElement = new Node<>(null, value, first.next);
+            first = newElement;
+            first.next.prev = newElement;
+            return element.value;
+        }
+        for (int i = 0; i < index; i++) {
+            element = element.next;
+        }
+        final T oldValue = element.value;
+        newElement = new Node<>(element.prev, value, element.next);
+        element.prev.next = newElement;
+        element.next.prev = newElement;
+        return oldValue;
```

Comment:
too complex logic, let's rewrite it: 
1. check the index
2. find node by index (for this purpose I recommend creat a separate method)
3. create a variable that will store the old value of the node by passed index
4. change value of node to new one
5. return old value